src/Components/WebAssembly/WebAssembly/src/Services/Logging/WebAssemblyConsoleLoggerProvider.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
+
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Services
+{
+    /// <summary>
+    /// A provider of <see cref="WebAssemblyConsoleLogger{T}"/> instances.
+    /// </summary>
+    public class WebAssemblyConsoleLoggerProvider : ILoggerProvider
+    {
+        private ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>> _loggers;
+        private IJSInProcessRuntime _jsRuntime;
+        private bool _disposed;
+
+        /// <summary>
+        /// Creates an instance of <see cref="WebAssemblyConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="WebAssemblyConsoleLogger"/> instances with.</param>
+        public WebAssemblyConsoleLoggerProvider(IJSInProcessRuntime jsRuntime)
+        {
+            _loggers = new ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>>();
+            _jsRuntime = jsRuntime;
+        }
+
+        /// <inheritdoc />
+        public ILogger CreateLogger(string name)
+        {
+            return _loggers.GetOrAdd(name, loggerName => new WebAssemblyConsoleLogger<object>(name, _jsRuntime));
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _loggers = null;
```

Comment:
Is there any chance we need to `Dispose` the `ILogger` instances before nulling this out?